pkg/operator/controller/gatewayclass/controller.go

```diff
@@ -0,0 +1,100 @@
+package gatewayclass
+
+import (
+	"context"
+
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+
+	"k8s.io/client-go/tools/record"
+
+	gatewayapiv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName = "gatewayclass_controller"
+
+	// OpenShiftGatewayClassControllerName is the string by which a
+	// gatewayclass identifies itself as belonging to OpenShift Istio.  If a
+	// gatewayclass's spec.controllerName field is set to this value, then
+	// the gatewayclass is ours.
+	OpenShiftGatewayClassControllerName = "openshift.io/gateway-controller"
```

Comment:
I thought I saw an "istio" gatewayclass on my cluster, and I thought I saw "istio" hard-coded in Istio.  Now that I check the Maistra code, I see that the default gatewayclass's name is configurable: https://github.com/maistra/istio/blob/maistra-2.4/pilot/pkg/features/pilot.go#L397-L398 (permalink: https://github.com/maistra/istio/blob/7b1d1a9b9be85e448ee8b235d1f5a52b8d09955b/pilot/pkg/features/pilot.go#L397-L398).  I don't think we can turn off creation of this default gatewayclass (the controller defined here always creates it if it doesn't already exist: https://github.com/maistra/istio/blob/maistra-2.4/pilot/pkg/config/kube/gateway/gatewayclass.go), but I can at least configure its name to be "ocp", or "openshift-default".  

I'd rather leave gatewayclass creation to the cluster admin, but in the absence of that option, I'll set `PILOT_GATEWAY_API_DEFAULT_GATEWAYCLASS=openshift-default`.  Does that seem reasonable?

Reply:
Done in https://github.com/openshift/cluster-ingress-operator/compare/c1c4ef672c6a88f64ad61f33ffaa30d83b993a44..503e0364c54c9f869b0ea8aaa7175b7bf458dca8.  Side-effect:  Now you need to specify `gatewayClassName: openshift-default` instead of `gatewayClassName: istio` on your gatewayclass.  :tada:!